### DISCUSSION TITLE
heat_exchanger with set_massflow

### DISCUSSION THREAD

**Discussion by ladderup** [e2nIEE/pandapipes#270] [Ideas]
In houses a heat_exchanger can controle the massflow depanding on the demand.
In the create_heat_exchanger() only the extracted qext_w can be set. The massflow is the result.

Setting the heat_exchanger massflow to zero will result in cooling of the district-heating-network in particuliere spots.
The cold spots can be fixed with a bypass.

I guess this can all be achieved with pipes, valves, pumps and heat_exchangers but still. A few more functions in a heat_exchanger could be nice.

**Comment by dcronbach** [comment]
Hallo,

Thanks for this comment. You are right that the heat exchanger functionalities are currently limited. We started to work on an improved version some time ago. Unfortunately, we have a also a strong focus on new features for simulating gas networks, which leads to a delay of new or improved components.

**Comment by jkisse** [comment]
There is a new pull request for this https://github.com/e2nIEE/pandapipes/pull/413
